### PR TITLE
Execution container extend

### PR DESCRIPTION
**Pull request by octocrock**
# About

Using execution containers, extending the octopusdeploy/worker-tools image and why you might create your own image.

# Pre-requisites

- [X] The draft is complete and this post is ready to be reviewed.
- [X] I'm familiar with the [writing for Octopus TL;DR](https://style.octopus.com/writing-at-octopus-tldr).
- [X] The PR build passes validation, and if it doesn't, I've checked the [common validations errors](https://style.octopus.com/writing-at-octopus-tldr#common-validation-errors) and none of those apply.
- [X] The images I've included follow the [rules for working with images](https://style.octopus.com/images#rules-for-screenshots).

## Publication date
Ideally before the release webinar on the 5th of August please :slightly_smiling_face:

### CODE REVIEW

**Comment by wordlee** [comment]
I've changed the order of the intro around a little bit to state first that this is new functionality, link to the previous blog post, and then the docs.

File: blog/2020-07/extending-octopus-execution-container/index.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
+---
+title: Execution containers for workers - Extending the Octopus worker-tools Docker image
```
Comment (wordlee) [inline blog/2020-07/extending-octopus-execution-container/index.md:2]:
```suggestion
title: "Execution containers for workers: Extending the Octopus worker-tools Docker image"
```

File: blog/2020-07/extending-octopus-execution-container/index.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+visibility: private
+published: 2999-01-01
+metaImage: 
+bannerImage: 
+tags:
+ - Octopus
+ - Product
+---
+
+We recently released [Execution Containers for Workers](https://octopus.com/blog/execution-containers), and we think there are a good number of reasons to use this new functionality. If you’re not familiar with execution containers, Michael does a great of introducing the concept in the post linked above, and you could also take a look at the [execution container documentation](https://g.octopushq.com/ExecutionContainersForWorkers).
```
Comment (wordlee) [inline blog/2020-07/extending-octopus-execution-container/index.md:14]:
```suggestion
We recently released [Execution Containers for Workers](https://octopus.com/blog/execution-containers), and we think there are a good number of reasons to use this new functionality. If you’re not familiar with execution containers, Michael does a great job of introducing the concept in the post linked above, and you could also take a look at the [execution container documentation](https://g.octopushq.com/ExecutionContainersForWorkers).
```

File: blog/2020-07/extending-octopus-execution-container/index.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+This way, you’re only getting PowerShell installed once, and you’re not adding bloat to the image, this is especially true should you want to specify different versions of other tools.
+
+## Size considerations
+
+The Octopus image covers many deployment scenarios, and it’s a solid choice as a starter image for your execution container.  Let’s take a look at the size of it:
+
+![](octo-worker-tools-size.png "width=500")
+
+It’s big, 3.44 GB.  And that’s the Linux flavor; the Windows image is larger, but in comparison to some images, it isn’t that big, but when you consider that you’re unlikely to use all the tools on that image, there’s a clear space saving to be made.
```
Comment (octocrock) [inline blog/2020-07/extending-octopus-execution-container/index.md:107]:
```suggestion
It’s big, 3.44 GB.  And that’s the Linux flavor; the Windows image is larger, but in comparison to some images, it isn’t that big, and when you consider that you’re unlikely to use all the tools on that image, there’s a clear space saving to be made.
```

**Pull request merged**